traffic_ops/testing/api/v4/jobs_test.go

```diff
@@ -360,3 +372,843 @@ func GetTestInvalidationJobs(t *testing.T) {
+	}
+
+	//Get Jobs by Invalid userID
+	opts = client.NewRequestOptions()
+	opts.QueryParameters.Set("userId", "11111")
+	toJobs, _, _ = TOSession.GetInvalidationJobs(opts)
+	if len(toJobs.Response) != 0 {
+		t.Errorf("Expected no response from Get Jobs by Invalid userID, but found %d ", len(toJobs.Response))
+	}
+}
+
+func CreateTestJobsInvalidds(t *testing.T) {
+	if len(testData.InvalidationJobs) < 1 {
+		t.Error("Need at least one Invalidation Jobs to test invalid ds")
+	}
+	job := testData.InvalidationJobs[0]
+	job.StartTime = &tc.Time{
+		Time:  time.Now().Add(time.Minute).UTC(),
+		Valid: true,
+	}
+	testData.InvalidationJobs[0] = job
+
+	//Invalid DS
+	request := tc.InvalidationJobInput{
+		DeliveryService: util.InterfacePtr("invalid"),
+		Regex:           job.Regex,
+		StartTime:       job.StartTime,
+		TTL:             job.TTL,
+	}
+	resp, reqInf, err := TOSession.CreateInvalidationJob(request, client.RequestOptions{})
+	if err == nil {
+		t.Errorf("Expected No DeliveryService exists matching identifier: %v - alerts: %v", request.DeliveryService, resp.Alerts)
+	}
+	if reqInf.StatusCode != http.StatusBadRequest {
+		t.Errorf("Expected status code 400, got %v", reqInf.StatusCode)
+	}
+
+	//Missing DS
+	request = tc.InvalidationJobInput{
+		Regex:     job.Regex,
+		StartTime: job.StartTime,
+		TTL:       job.TTL,
+	}
+	resp, reqInf, err = TOSession.CreateInvalidationJob(request, client.RequestOptions{})
+	if err == nil {
+		t.Errorf("Expected deliveryService: cannot be blank - alerts: %v", resp.Alerts)
+	}
+	if reqInf.StatusCode != http.StatusBadRequest {
+		t.Errorf("Expected status code 400, got %v", reqInf.StatusCode)
+	}
+
+	//Empty DS
+	request = tc.InvalidationJobInput{
+		DeliveryService: util.InterfacePtr(""),
+		Regex:           job.Regex,
+		StartTime:       job.StartTime,
+		TTL:             job.TTL,
+	}
+	resp, reqInf, err = TOSession.CreateInvalidationJob(request, client.RequestOptions{})
+	if err == nil {
+		t.Errorf("Expected deliveryService: cannot be blank., No DeliveryService exists matching identifier: - alerts: %v", resp.Alerts)
+	}
+	if reqInf.StatusCode != http.StatusBadRequest {
+		t.Errorf("Expected status code 400, got %v", reqInf.StatusCode)
+	}
+}
+
+func CreateTestJobsAlreadyExistTTL(t *testing.T) {
+	if len(testData.InvalidationJobs) < 1 {
+		t.Error("Need at least one Invalidation Jobs to create duplicate data")
+	}
+	job := testData.InvalidationJobs[0]
+	job.StartTime = &tc.Time{
+		Time:  time.Now().Add(time.Minute).UTC(),
+		Valid: true,
+	}
+	testData.InvalidationJobs[0] = job
+
+	request := tc.InvalidationJobInput{
+		DeliveryService: job.DeliveryService,
+		Regex:           job.Regex,
+		StartTime:       job.StartTime,
+		TTL:             job.TTL,
+	}
+	resp, _, err := TOSession.CreateInvalidationJob(request, client.RequestOptions{})
+	if err != nil {
+		t.Errorf("Expected Invalidation request created, but found error %v - Alert %v", err, resp.Alerts)
```

Comment:
Umm.. is the expectation to create the job request? or not to, since the job already exists?

Reply:
yes, this is the expected error message.

Reply:
So, we are ok to create duplicate jobs for a same DS with exact start and TTL times?

Reply:
yes, we have a test case to create with this scenario. If it's not the expected scenario, I will deprecate this test.

Reply:
No need to deprecate. I was merely trying to understand why would we create two jobs with same start and TTL times belonging to one DS.